engine/src/main/java/org/terasology/rendering/world/WorldRendererImpl.java

```diff
@@ -618,45 +608,50 @@ private void renderSimpleBlendMaterials() {
         PerformanceMonitor.endActivity();
     }
 
-    private void renderChunk(RenderableChunk chunk, ChunkMesh.RenderPhase phase, Camera camera, ChunkRenderMode mode) {
```

Comment:
(re-adding this comment,  because I changed the target line and force-pushed)
The diff does not do these changes justice. Basically I moved up all the stuff that is the same for all the chunks (camera position, logic about render phase, and common uniforms), and then made sure all uniform setting used the "currentOnly = true".

Most importantly moved up the chunkShader.activateFeature(...) because that causes the shader to become dirty and re-enable the shader with openGL.